frontend/packages/pipelines-plugin/src/components/repository/conts.ts

```diff
@@ -0,0 +1,23 @@
+interface RepositoryLabel {
+  labels: string[];
+  baseURL?: string;
+}
+
+export enum REPOSITORY_FIELDS {
+  REPOSITORY = 'Repository',
+  BRANCH = 'Branch',
+  REPO_URL = 'Repo url',
+  SHA = 'sha',
+  EVENT_TYPE = 'Event type',
+}
+
+export const repositoryLabels: Record<REPOSITORY_FIELDS, RepositoryLabel> = {
+  [REPOSITORY_FIELDS.REPOSITORY]: { labels: ['pipelinesascode.tekton.dev/repository'] },
```

Comment:
Save all the required repository labels in an enum `RepositoryLabels` 

```
export enum RepositoryLabels {
    Repository = 'pipelinesascode.tekton.dev/repository',
    Branch = 'pipelinesascode.tekton.dev/branch',
    RepoUrl = 'pipelinesascode.tekton.dev/url-repository', 
    RepoOrg = 'pipelinesascode.tekton.dev/url-org'
    Sha = 'pipelinesascode.tekton.dev/sha',
    EventType = 'pipelinesascode.tekton.dev/event-type',
}
```

Reply:
I am now using a single data structure, but I dont want to use string based keys as these would be used in multiple places